src/examples/token-manager.new.ts

```diff
@@ -0,0 +1,380 @@
+import {
+  Field,
+  ZkProgram,
+  Struct,
+  Bool,
+  Bytes,
+  Hash,
+  SmartContract,
+  method,
+  Permissions,
+  DeployArgs,
+  Crypto,
+  createForeignCurve,
+  createEcdsa,
+  PublicKey,
+  State,
+  state,
+  UInt64,
+  VerificationKey,
+  AccountUpdate,
+  Mina,
+  PrivateKey,
+  UInt8,
+} from 'o1js';
+import { FungibleToken, VKeyMerkleMap } from '../NewTokenStandard.js';
+import {
+  generateDummyDynamicProof,
+  SideloadedProof,
+} from '../side-loaded/program.eg.js';
+import {
+  BurnConfig,
+  BurnDynamicProofConfig,
+  BurnParams,
+  MintConfig,
+  MintDynamicProofConfig,
+  MintParams,
+  TransferDynamicProofConfig,
+  UpdatesDynamicProofConfig,
+} from '../configs.js';
+import { equal } from 'node:assert';
+
+// === ECDSA ZkProgram Definition ===
+
+export class Secp256k1 extends createForeignCurve(
+  Crypto.CurveParams.Secp256k1
+) {}
+export class Ecdsa extends createEcdsa(Secp256k1) {}
```

Comment:
Do you think we should export this straight from o1js?  We write these lines over and over in lots of examples!

Reply:
While most examples use secp256k1 for ECDSA, users might want to work with other curves like secp256r1, and having it defined explicitly gives them a clear template for customization. 

We could consider exporting a default like Secp256k1Ecdsa for convenience, while still encouraging customization when needed. What do you think?